### PR TITLE
[Release WDL] Split draft from publish to confirm creds early

### PR DESCRIPTION
**Pull request by cjllanwarne**
By making the draft release before building Cromwell, we will find out early if there's a problem with the auth token

### CODE REVIEW

File: release/release_workflow.wdl [hunk 1]
```diff
@@ -124,25 +124,19 @@ task versionPrep {
...
-    String cromwellJarName = basename(cromwellJar)
-    String womtoolJarName = basename(womtoolJar)
-
     command <<<
         set -e
         set -x
 
         # download changelog from master
-        curl https://raw.githubusercontent.com/~{organization}/cromwell/master/CHANGELOG.md -o CHANGELOG.md
+        curl --fail https://raw.githubusercontent.com/~{organization}/cromwell/master/CHANGELOG.md -o CHANGELOG.md
```
Comment (aednichols) [inline release/release_workflow.wdl]: [ToL] It looks like `--fail` has the side effect of suppressing error output - could make things harder to debug (last time, we figured out the problem from the output).

But `--fail` >> nothing so 🤷‍♂️

File: release/release_workflow.wdl [hunk 2]
```diff
@@ -308,46 +334,57 @@ workflow release_cromwell {
...
-    }
+  if (draftGithubRelease.draft_complete) {
+      if (majorRelease) {
+        call do_major_release { input:
+                organization = organization,
+                releaseVersion = cromwellVersion
+        }
+      }
+
+      if (!majorRelease) {
```
Comment (Horneth) [inline release/release_workflow.wdl]: When is `else` coming ?! 😄

Comment (mcovarr) [reply release/release_workflow.wdl]: 🔥

File: release/release_workflow.wdl [hunk 3]
```diff
@@ -113,8 +106,10 @@ task versionPrep {
     }
 
     command <<<
+      set -e
+
       which jq || brew install jq
-      curl -s https://api.github.com/repos/~{organization}/cromwell/releases/latest | jq --raw-output '.tag_name' > version
+      curl --fail -v -s https://api.github.com/repos/~{organization}/cromwell/releases/latest | jq --raw-output '.tag_name' > version
```
Comment (mcovarr) [inline release/release_workflow.wdl]: what's the net effect of `--fail` (fail silently) and `-v` (be verbose)?

Comment (cjllanwarne) [reply release/release_workflow.wdl]: - `--fail` forces curl to change its exit code if something goes wrong
- `-v` means that we don't lose all context if that happens

**Pull request merged**